synapseclient/models/dataset.py

```diff
@@ -1618,13 +1660,1369 @@ async def snapshot_async(
+        )
+
+    def fill_from_dict(self, entity, set_annotations: bool = True) -> "Self":
+        """
+        Converts the data coming from the Synapse API into this datamodel.
+
+        Arguments:
+            synapse_table: The data coming from the Synapse API
+
+        Returns:
+            The DatasetCollection object instance.
+        """
+        self.id = entity.get("id", None)
+        self.name = entity.get("name", None)
+        self.description = entity.get("description", None)
+        self.parent_id = entity.get("parentId", None)
+        self.etag = entity.get("etag", None)
+        self.created_on = entity.get("createdOn", None)
+        self.created_by = entity.get("createdBy", None)
+        self.modified_on = entity.get("modifiedOn", None)
+        self.modified_by = entity.get("modifiedBy", None)
+        self.version_number = entity.get("versionNumber", None)
+        self.version_label = entity.get("versionLabel", None)
+        self.version_comment = entity.get("versionComment", None)
+        self.is_latest_version = entity.get("isLatestVersion", None)
+        self.is_search_enabled = entity.get("isSearchEnabled", False)
+        self.items = [
+            EntityRef(id=item["entityId"], version=item["versionNumber"])
+            for item in entity.get("items", [])
+        ]
+        if set_annotations:
+            self.annotations = Annotations.from_dict(entity.get("annotations", {}))
+        return self
+
+    def to_synapse_request(self):
+        """Converts the request to a request expected of the Synapse REST API."""
+
+        entity = {
+            "name": self.name,
+            "description": self.description,
+            "id": self.id,
+            "etag": self.etag,
+            "createdOn": self.created_on,
+            "modifiedOn": self.modified_on,
+            "createdBy": self.created_by,
+            "modifiedBy": self.modified_by,
+            "parentId": self.parent_id,
+            "concreteType": concrete_types.DATASET_COLLECTION_ENTITY,
+            "versionNumber": self.version_number,
+            "versionLabel": self.version_label,
+            "versionComment": self.version_comment,
+            "isLatestVersion": self.is_latest_version,
+            "columnIds": (
+                [
+                    column.id
+                    for column in self._last_persistent_instance.columns.values()
+                ]
+                if self._last_persistent_instance
+                and self._last_persistent_instance.columns
+                else []
+            ),
+            "isSearchEnabled": self.is_search_enabled,
+            "items": (
+                [item.to_synapse_request() for item in self.items] if self.items else []
+            ),
+        }
+        delete_none_keys(entity)
+        result = {
+            "entity": entity,
+        }
+        delete_none_keys(result)
+        return result
+
+    async def add_item_async(
```

Comment:
Do `remove_item_async` and this `add_item_async` need to be an async method? I don't see that it is internally calling any async functions.

This could work similar to how we are doing these functions: https://python-docs.synapse.org/en/latest/reference/experimental/async/table/#synapseclient.models.Table.add_column